azurerm/resource_arm_storage_account_network_rules.go

```diff
@@ -0,0 +1,275 @@
+package azurerm
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/storage/mgmt/2019-04-01/storage"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/locks"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+var storageAccountResourceName = "azurerm_storage_account"
+
+func resourceArmStorageAccountNetworkRules() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmStorageAccountNetworkRulesCreateUpdate,
+		Read:   resourceArmStorageAccountNetworkRulesRead,
+		Update: resourceArmStorageAccountNetworkRulesCreateUpdate,
+		Delete: resourceArmStorageAccountNetworkRulesDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"storage_account_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateArmStorageAccountName,
+			},
+
+			"bypass": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+					ValidateFunc: validation.StringInSlice([]string{
+						string(storage.AzureServices),
+						string(storage.Logging),
+						string(storage.Metrics),
+						string(storage.None),
+					}, true),
+				},
+				Set: schema.HashString,
+			},
+
+			"ip_rules": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
```

Comment:
as this is an opt computed block we should use the weird config atter mode hack so it can be unset?

Reply:
I don't believe I've seen this

Reply:
Search for `ConfigMode: schema.SchemaConfigModeAttr,`, its basically to force tf to allow array syntax in .12 because otherwise there is no way to send an empty list to clear the property.

Reply:
Thank you thank you